pkg/scale/errors.go

```diff
@@ -6,5 +6,17 @@ package scale
 import "errors"
 
 var (
-	ErrVaryingDataTypeNotSet = errors.New("VaryingDataTypeValue has not been set")
+	ErrUnsupportedDestination            = errors.New("must be a pointer to a destination, unsupported destination")
+	errDecodeBool                        = errors.New("decodeBool failed")
+	ErrUnsupportedType                   = errors.New("unsupported type")
+	errUnsupportedResult                 = errors.New("unsupported result")
+	errUnsupportedOption                 = errors.New("unsupported option")
+	errFindVDT                           = errors.New("unable to find VaryingDataTypeValue with index")
+	errUint128IsNil                      = errors.New("uint128 in nil")
+	errResultNotSet                      = errors.New("result not set")
```

Comment:
```suggestion
	ErrResultNotSet                      = errors.New("result not set")
```
I could argue this one should be exposed.